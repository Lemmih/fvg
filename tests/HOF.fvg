apply : (a -> b) -> (a -> b)
apply f a = f a

dot : (b -> c) -> (a -> b) -> a -> c
dot f g x = f (g x)

main : Svg
main =
  <svg>
    <text "font-size"="35">
      apply (plusI 1) 2
    </text> ;
    <text y="35" "font-size"="35">
      dot (plusI 1) (plusI 10) 5
    </text>
  </svg>
