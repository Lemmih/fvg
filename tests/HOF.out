 <svg > <text font-size="35" >3 </text>  <text y="35" font-size="35" >16 </text>  </svg> 